crates/sql_provider_datafusion/src/dbconnectionpool/postgrespool.rs

```diff
@@ -42,7 +42,30 @@ impl
         _mode: Mode,
         _params: Arc<Option<HashMap<String, String>>>,
     ) -> Result<Self> {
-        let connection_string = "host=localhost user=postgres password=postgres dbname=postgres";
+        let mut host = "localhost";
+        let mut user = "postgres";
+        let mut dbname = "postgres";
+        let mut connection_string = String::new();
+
+        if let Some(params) = _params.as_ref() {
+            if let Some(pg_host) = params.get("pg_host") {
+                host = pg_host;
+            }
+            if let Some(pg_user) = params.get("pg_user") {
+                user = pg_user;
+            }
+            if let Some(pg_db) = params.get("pg_db") {
+                dbname = pg_db;
+            }
+            if let Some(pg_pass) = params.get("pg_pass") {
```

Comment:
We need to support @ewgenius work on secrets for providing the password

Reply:
Thanks, will update the changes